example_12/controllers/src/passthrough_controller.cpp

```diff
@@ -0,0 +1,25 @@
+// Copyright (c) 2023, PAL Robotics
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "passthrough_controller/passthrough_controller.hpp"
+
+namespace passthrough_controller
+{
+
```

Comment:
Is there a special reason why you defined the functions in the header file, if you have already a c++ file too?

Reply:
Well, I used it as the convention used in the ros2_controllers repo as well, moreover, I like to separate definitions and declarations into separate files. If you don't like this, I can merge them into .cpp

Reply:
Inside this repo it is not very important because we don't release them and the header file itself won't be used by any other user.
I don't have a clear opinion against it, but which ros2_controller has this structure? Imho, all methods are defined in its' cpp source files.

Reply:
Well, I can remove it, if needed. I don't have a problem with it.

You cna find the similar structure with, [joint_state_broadcaster](https://github.com/ros-controls/ros2_controllers/blob/master/joint_state_broadcaster/include/joint_state_broadcaster/joint_state_broadcaster.hpp), [admittance_controller](https://github.com/ros-controls/ros2_controllers/blob/master/admittance_controller/include/admittance_controller/admittance_controller.hpp) etc..

Let me know :)

Reply:
Are we talking about the same? :D What I meant is that you have defined all your functions in the header file, and the cpp file is almost empty, having only the pluginlib export macro.

Reply:
Shit! You are absolutely right. I'm under the wrong impression. I think this needs to be fixed before merging the PR. I will do it sometime today.
Sorry for all the mess. 

Thank you

Reply:
Updated!. Thanks for the valuable feedback.